### ISSUE TITLE
rgbaRegex fails 3 digit decimals

### ISSUE THREAD

**Issue by levymetal** [styled-components/polished#610]
## Summary

`polished` version: 4.2.1

After https://github.com/styled-components/polished/pull/609, `rgbaRegex` no longer allows more than two decimal places for the alpha component. While this is an edge case, it's not totally unreasonable - in theory the minimum step should be 1/256. Also, because it now accepts a percentage, the maximum precision differs depending on the unit. Eg, we can pass `2.8%` but not `0.028`.

I'm happy to whip up a PR if this is considered a regression - just let me know.

## Mixin/Helper/Shorthand Usage

```javascript
readableColor('rgba(243,112,33,0.028)');
```

## What You Are Seeing

`Couldn't parse the color string. Please provide the color as a string in hex, rgb, rgba, hsl or hsla notation.`

## What You Expected To See

`#000`

## Reproduction

n/a

**Comment by jrmyio** [comment]
@bhough  this is definitely a pretty nasty bug:

Since version: 4.2.1:
```ts
// Throws no error
console.log(shade(0.05, 'rgba(247,213,175,0.56)'));

// Throws error since 4.2.1
console.log(mix(0.7, shade(0.05, 'rgba(247,213,175,0.56)'), '#3768fa'));
```

Codesandbox:
https://codesandbox.io/s/upbeat-bush-9ylp0k?file=/src/index.js:0-166

**Comment by bhough** [comment]
This should be fully addressed in [4.2.2](https://github.com/styled-components/polished/releases/tag/v4.2.2)
